Source/i_sound.h

```diff
@@ -90,39 +92,59 @@ int I_SoundID(int handle);
 //  MUSIC I/O
 //
 
+typedef struct
+{
+    boolean (*I_InitMusic)(void);
+    void (*I_ShutdownMusic)(void);
+    void (*I_SetMusicVolume)(int volume);
+    void (*I_PauseSong)(void *handle);
+    void (*I_ResumeSong)(void *handle);
+    void *(*I_RegisterSong)(void *data, int size);
+    void (*I_PlaySong)(void *handle, boolean looping);
+    void (*I_StopSong)(void *handle);
+    void (*I_UnRegisterSong)(void *handle);
+} music_module_t;
+
 typedef enum
 {
-  music_backend_sdl,
-  music_backend_opl,
-  num_music_backends,
-} music_backend_t;
+#if defined(_WIN32)
```

Comment:
No SDL backend on WIndows? How do we playback MP3 music then?

Reply:
The SDL music backend is now always initialized, `I_RegisterSong()` calls the SDL module for non-MIDI music. See `I_RegisterSong` function in `i_sound.c`.

Reply:
👍

Reply:
Wait a minute! Does this mean we can play back MIDI music with OPL emulation and the next song is in MP3 format and we can play it back with SDL? How cool is that?!

Reply:
Yes, we can. Maybe we should add this to Crispy :smile: